internal/tools/observability/cloudmonitoring/cloudmonitoring.go

```diff
@@ -0,0 +1,179 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudmonitoring
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/goccy/go-yaml"
+	"github.com/googleapis/genai-toolbox/internal/sources"
+	"github.com/googleapis/genai-toolbox/internal/tools"
+	"golang.org/x/oauth2/google"
+)
+
+var (
+	googleFindDefaultCredentials = google.FindDefaultCredentials
+	monitoringEndpoint           = "https://monitoring.googleapis.com"
+)
+
+// SetMonitoringEndpoint sets the monitoring endpoint for testing purposes.
+func SetMonitoringEndpoint(endpoint string) {
+	monitoringEndpoint = endpoint
+}
+
+// SetGoogleFindDefaultCredentials sets the google.FindDefaultCredentials function for testing purposes.
+func SetGoogleFindDefaultCredentials(f func(ctx context.Context, scopes ...string) (*google.Credentials, error)) {
+	googleFindDefaultCredentials = f
+}
+
+const kind string = "cloudmonitoring"
+
+func init() {
+	if !tools.Register(kind, newConfig) {
+		panic(fmt.Sprintf("tool kind %q already registered", kind))
+	}
+}
+
+func newConfig(ctx context.Context, name string, decoder *yaml.Decoder) (tools.ToolConfig, error) {
+	actual := Config{Name: name}
+	if err := decoder.DecodeContext(ctx, &actual); err != nil {
+		return nil, err
+	}
+	return actual, nil
+}
+
+type Config struct {
+	Name         string   `yaml:"name" validate:"required"`
+	Kind         string   `yaml:"kind" validate:"required"`
+	Description  string   `yaml:"description" validate:"required"`
+	Params       tools.Parameters `yaml:"params"`
+	AuthRequired []string `yaml:"authRequired"`
+}
+
+// validate interface
+var _ tools.ToolConfig = Config{}
+
+func (cfg Config) ToolConfigKind() string {
+	return kind
+}
+
+func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error) {
+	return Tool{
+		Name:        cfg.Name,
+		Kind:        kind,
+		Description: cfg.Description,
+		Params:      cfg.Params,
+		manifest:    tools.Manifest{Description: cfg.Description, Parameters: cfg.Params.Manifest()},
+		mcpManifest: tools.McpManifest{Name: cfg.Name, Description: cfg.Description, InputSchema: cfg.Params.McpManifest()},
+	}, nil
+}
+
+// validate interface
+var _ tools.Tool = Tool{}
+
+type Tool struct {
+	Name        string   `yaml:"name"`
+	Kind        string   `yaml:"kind"`
+	Description string   `yaml:"description"`
+	Params      tools.Parameters `yaml:"params"`
+	manifest    tools.Manifest
+	mcpManifest tools.McpManifest
+}
+
+func (t Tool) Invoke(ctx context.Context, params tools.ParamValues, accessToken tools.AccessToken) (any, error) {
+	paramsMap := params.AsMap()
+	projectID, ok := paramsMap["projectID"].(string)
+	if !ok {
+		return nil, fmt.Errorf("projectID parameter not found or not a string")
+	}
+	query, ok := paramsMap["query"].(string)
```

Comment:
We could think about providing a hard coded input for specific metrics or move the query description to the tool description